### ISSUE TITLE
Ansible Vault Python 3 String Error

### ISSUE THREAD

**Issue by naftulikay** [ansible/ansible#21763]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Ansible Vault.

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file = /home/naftuli/Documents/Development/naftulikay/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

```
[defaults]
ask_sudo_pass = true
command_warnings = false
hash_behavior = merge
inventory = inventory/
retry_files_enabled = false
roles_path = roles:galaxy_roles
vault_password_file = bin/decrypt-vault
```

##### OS / ENVIRONMENT

elementary Loki, being Ubuntu 16.04, kernel 4.4.0-62-generic.

##### SUMMARY

Exception thrown when decrypting the vault password file:

```
ERROR! Unexpected Exception: a bytes-like object is required, not 'str'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python3.5/dist-packages/ansible/cli/playbook.py", line 110, in run
    vault_pass = CLI.read_vault_password_file(self.options.vault_password_file, loader=loader)
  File "/usr/local/lib/python3.5/dist-packages/ansible/cli/__init__.py", line 576, in read_vault_password_file
    vault_pass = stdout.strip('\r\n')
TypeError: a bytes-like object is required, not 'str'
```

Ansible isn't handling the strings properly in this case for Python 3.

##### STEPS TO REPRODUCE

My vault script is very simple:

```
#!/usr/bin/env bash

gpg2 --batch --decrypt vault-passphrase.gpg
```

It is called via my Ansible configuration above.

##### EXPECTED RESULTS

When Ansible is installed via `pip2`, it works; the password file is decrypted and provisioning occurs as expected.

##### ACTUAL RESULTS

The following exception is thrown:

```
ERROR! Unexpected Exception: a bytes-like object is required, not 'str'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python3.5/dist-packages/ansible/cli/playbook.py", line 110, in run
    vault_pass = CLI.read_vault_password_file(self.options.vault_password_file, loader=loader)
  File "/usr/local/lib/python3.5/dist-packages/ansible/cli/__init__.py", line 576, in read_vault_password_file
    vault_pass = stdout.strip('\r\n')
TypeError: a bytes-like object is required, not 'str'
```

**Comment by naftulikay** [comment]
This issue is similar to what I reported before in the Route 53 module, which was fixed in eb7495efaf73a0858a8ebd1a32463bf92fee9e90. That issue was reported at #21027.

**Comment by naftulikay** [comment]
The exact line throwing the error [is found here](https://github.com/ansible/ansible/blob/v2.2.0.0-1/lib/ansible/cli/__init__.py#L576).

**Comment by abadger** [comment]
This was fixed in ansible-2.2.1 and then fixed in a different way in the soon to be released ansible-2.2.2 (because the 2.2.1 fix has difficulties with passwords encoded in non-utf8).  Could you test with 2.2.1 or the 2.2.2 release candidate to see if this is now fixed for you?  Both can be found on http://releases.ansible.com/ansible/

needs_info

**Comment by naftulikay** [comment]
@abadger I have tried it out with the newer 2.2.1 and it is fixed. I am closing as fixed by that version.
